gateway/admin.go

```diff
@@ -21,3 +32,170 @@ func (g *GatewayAdmin) Status() interface{} {
 	}
 
 }
+
+type GatewayRPCHandler struct {
+	jobsDB jobsdb.JobsDB
+}
+
+type SqlRunner struct {
+	dbHandle     *sql.DB
+	jobTableName string
+}
+
+type SourceEvents struct {
+	Count  int
+	Source string
+}
+
+func (r *SqlRunner) getUniqueSources() ([]SourceEvents, error) {
+	var rows *sql.Rows
+	var err error
+	sources := make([]SourceEvents, 0)
+	uniqueSourceValsStmt := fmt.Sprintf(`select count(*) as count, parameters -> 'source_id' as source from %s group by parameters -> 'source_id';`, r.jobTableName)
+	rows, err = r.dbHandle.Query(uniqueSourceValsStmt)
+	if err != nil {
+		return sources, err
+	}
+	defer rows.Close()
+	sourceEvent := SourceEvents{}
+	for rows.Next() {
+		err = rows.Scan(&sourceEvent.Count, &sourceEvent.Source)
+		if err != nil {
+			return sources, err // defer closing of rows, so return will not memory leak
+		}
+		sources = append(sources, sourceEvent)
+	}
+
+	err = rows.Err()
```

Comment:
Follow similar pattern with what we are doing.

Reply:
done